src/app/services/tests/GameService.spec.ts

```diff
@@ -232,105 +228,51 @@ describe('GameService', () => {
-                partStatus: 3,
-                partType: PartType.BLITZ.value,
-                totalPartDuration: 25,
-            };
-            await receiveRematchProposal(Player.ZERO);
+            // Given a previous game
+            const lastPart: PartDocument = new PartDocument('partId', PartMocks.FINISHED);
+            const lastGameConfigRoom: ConfigRoom = ConfigRoomMocks.WITH_ACCEPTED_CONFIG;
             spyOn(configRoomService, 'readConfigRoomById').and.resolveTo(lastGameConfigRoom);
-            let called: boolean = false;
-            spyOn(partDAO, 'set').and.callFake(async(_id: string, element: Part) => {
-                expect(element.playerZero).toEqual(Utils.getNonNullable(lastPart.data.playerOne));
-                expect(element.playerOne).toEqual(Utils.getNonNullable(lastPart.data.playerZero));
-                called = true;
-            });
+            spyOn(partDAO, 'create').and.resolveTo('rematchId');
 
-            // When accepting rematch as Player.ONE
+            // When calling acceptRematch
             await gameService.acceptRematch(lastPart, Player.ONE);
 
-            // Then we should have a part created with playerOne and playerZero switched
-            expect(called).toBeTrue();
+            // Then it should create a new part with the players reversed
+            const part: Part = {
+                ...PartMocks.STARTED,
+                playerZero: Utils.getNonNullable(lastPart.data.playerOne),
+                playerOne: lastPart.data.playerZero,
+            };
+            expect(partDAO.create).toHaveBeenCalledOnceWith(part);
         }));
         it('should start with the other player when first player was random', fakeAsync(async() => {
-            // Given a previous match with creator starting, and creator proposes a rematch
-            const lastPart: PartDocument = new PartDocument('partId', {
-                playerZero: UserMocks.OPPONENT_MINIMAL_USER,
-                playerOne: UserMocks.CREATOR_MINIMAL_USER,
-                result: MGPResult.VICTORY.value,
-                turn: 2,
-                typeGame: 'laMarelle',
-                beginning: new Timestamp(1700102, 680000000),
-                loser: UserMocks.CREATOR_MINIMAL_USER,
-                winner: UserMocks.OPPONENT_MINIMAL_USER,
-            });
-            const lastGameConfigRoom: ConfigRoom = {
-                chosenOpponent: UserMocks.OPPONENT_MINIMAL_USER,
-                creator: UserMocks.CREATOR_MINIMAL_USER,
-                firstPlayer: 'RANDOM',
-                maximalMoveDuration: 10,
-                partStatus: 3,
-                partType: PartType.BLITZ.value,
-                totalPartDuration: 25,
-            };
-            await receiveRematchProposal(Player.ZERO);
+            // Given a previous game
+            const lastPart: PartDocument = new PartDocument('partId', PartMocks.FINISHED);
+            const lastGameConfigRoom: ConfigRoom = ConfigRoomMocks.WITH_ACCEPTED_CONFIG;
```

Comment:
en vrai là, vu que "random or not" est pas pris en compte pour choisir le prochain joueur, j'serais d'humeur à dire qu'on peut remover ce test !